include/deal.II/grid/tria_iterator.templates.h

```diff
@@ -132,18 +132,7 @@ inline
 bool
 TriaRawIterator<Accessor>::operator != (const TriaRawIterator<Accessor> &i) const
 {
-  // Note that at times, there is a problem
-  // with egcs 1.1 that makes it choose
-  // the global STL operator != (which
-  // does only !(a==b)) over the member
-  // function one, which then results in an
-  // error because the operator == of
-  // the accessor class is
-  // not made public. Strange... don't know
-  // whose fault it is.
-  //
-  // Work around the problem this way:
-  return accessor.operator != (i.accessor);
+  return not (*this == i);
```

Comment:
This is a good point: deal.II consistently uses `!`, `&&`, and `||` instead of `not`, `and`, and `or` (except for this line). I will look into another PR that addresses this.